static/src/js/components/Map/ShapefileLayer.js

```diff
@@ -105,10 +119,22 @@ class ShapefileLayerExtended extends L.Layer {
     })
   }
 
-  onSuccess(file, response) {
+  onSuccess(file, response, panMap = true) {
     // look through response and separate all MultiPolygon types into their own polygon
     this.separateMultiPolygons(response)
 
+    // Add an id field to each feature unless it already has one
+    response.features.map((feature, index) => {
+      const newFeature = feature
+      const { id } = feature
+
+      if (!id) {
+        newFeature.id = index.toString()
```

Comment:
change newFeature.id to newFeature.edscId to ensure we don't conflict with existing IDs